src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -305,3 +309,108 @@ def query(self):
                  on the API documentation that was not implemented yet.
         """
         pass
+
+
+class ZFSQuoteService(Service):
+
+    class Config:
+        namespace = 'zfs.quota'
+        private = True
+
+    @periodic(60)
+    async def notify_quota_excess(self):
+        existing_excesses = {
+            excess["dataset_name"]: excess
+            for excess in await self.middleware.call('datastore.query', 'storage.quotaexcess')
+        }
+
+        await self.middleware.call('datastore.sql', 'UPDATE storage_quotaexcess SET level = -1')
```

Comment:
Writing so frequently to the database is something we try to avoid at all costs.

Most of FreeNAS installations use cheap USB flash drives as boot media, so updating the database every 60 seconds would wear out these devices very quickly.

Is there something we could do here to not do that?
Perhaps use a semi-persistent file in /tmp (or even use the cache middleware plugin)? That means this data would not persist across reboots of course, but I would think thats an acceptable trade offer.

Reply:
1. Write to file in /tmp periodically.
2. Save latest value to DB on reboot/shutdown.
3. Pull values from DB on startup/boot back into file in /tmp.
4. Start back at one.

(just throwing in my two cents)

:)